server/clip_server/model/clip_model.py

```diff
@@ -5,7 +5,25 @@
 )
 
 
-class CLIPModel:
+class BaseCLIPModel:
+    def __init__(self, name: str, **kwargs):
+        super().__init__()
+        self._name = name
+
+    @staticmethod
+    def get_model_name(name: str):
+        return name
+
+    @property
+    def model_name(self):
+        return self.__class__.get_model_name(self._name)
```

Comment:
can we simply return `xxx.name`?

Reply:
To get the model name without initializing the instance (avoiding downloading and loading weights), we must use *staticmethod* . That's the reason I implement `get_model_name`